web/client/utils/FileUtils.js

```diff
@@ -23,14 +32,55 @@ const FileUtils = {
     shpToGeoJSON: function(zipBuffer) {
         return [].concat(shp.parseZip(zipBuffer));
     },
+    kmlToGeoJSON: function(xml) {
+        let geoJSON = [].concat(tj.kml(xml));
+        geoJSON.forEach(function(item) {
+            item.fileName = xml.getElementsByTagName('name')[0].innerHTML;
+        });
+        return geoJSON;
+    },
+    gpxToGeoJSON: function(xml) {
+        let geoJSON = [].concat(tj.gpx(xml));
+        geoJSON.forEach(function(item) {
+            item.fileName = xml.getElementsByTagName('name')[0].innerHTML;
+        });
+        return geoJSON;
+    },
     readZip: function(file) {
         return new Promise((resolve, reject) => {
             let reader = new FileReader();
             reader.onload = function() { resolve(reader.result); };
             reader.onerror = function() { reject(reader.error.name); };
             reader.readAsArrayBuffer(file);
         });
-
+    },
+    readKml: function(file) {
+        return new Promise((resolve, reject) => {
+            let reader = new FileReader();
+            reader.onload = function() {
+                 resolve(parser.parseFromString(reader.result, "text/xml"));
+             };
+            reader.onerror = function() {
+                reject(reader.error.name);
+            };
+            reader.readAsText(file);
+        });
+    },
+    readKmz: function(file) {
+        const zip = new JSZip();
+        return new Promise((resolve, reject) => {
+            zip.loadAsync(file).then(files => {
+                files.filter(elem => {
+                    return elem.indexOf('kml') !== -1;
+                }).map(elem => {
```

Comment:
Here you have to use forEach, instead, because you have side effects, it doesn't make sense to use map if you ignore the resulting array